ppa/archive/management/commands/hathi_rsync.py

```diff
@@ -40,52 +42,88 @@ def handle(self, *args, **kwargs):
+            source_ids=working_htids, rsync_output=True, output_dir=output_dir.name
         )
         logfile = htimporter.rsync_data()
 
-        # read the rsync itemized output to identify records where file
-        # sizes changed
-        updated_ids = set()
+        # read the rsync itemized output to identify and report on changes
+        updated_files = []
         with open(logfile) as rsync_output:
             for line in rsync_output:
-                # if a line indicates that a file was updated due
-                # to a change in size, use the path to determine the hathi id
-                if " >f.s" in line:
-                    # rsync itemized output is white-space delimited;
+                # check for a line indicating that a file was updated
+                if " >f" in line:
+                    # rsync itemized output is white-space delimited
+                    parts = line.split()
                     # last element is the filename that was updated
-                    filename = line.rsplit()[-1].strip()
+                    filename = parts[-1]
+                    # itemized info flags preced the filename
+                    flags = parts[-2]
+
                     # we only care about zip files and mets.xml files
                     if not filename.endswith(".zip") and not filename.endswith(".xml"):
                         continue
                     # reconstruct the hathi id from the filepath
-                    ht_prefix, pairtree_dir = filename.split("/pairtree_root/")
+                    ht_prefix, pairtree_dir = filename.split("/pairtree_root/", 1)
                     # get the directory one level up from the updated file
                     pairtree_id = os.path.dirname(os.path.dirname(pairtree_dir))
                     # use pairtree to determine the id based on the path
                     # (handles special characters like those used in ARKs)
                     htid = f"{ht_prefix}.{path2id(pairtree_id)}"
-                    updated_ids.add(htid)
+                    updated_files.append(
+                        {
+                            "htid": htid,
+                            "filename": os.path.basename(filename),
+                            # rsync itemized flags look like >f.st....
+                            # or >f+++++++ for new files
+                            "size_changed": flags[3] == "s",
+                            "modification_time": flags[4] == "t",
+                            "rsync_flags": flags,
+                        }
+                    )
 
         # should this behavior only be when updating all?
         # if specific htids are specified on the command line, maybe report on them only?
-        if updated_ids:
-            outfilename = "ppa_rsync_updated_htids_%s.txt" % datetime.now().strftime(
+        if updated_files:
+            outfilename = "ppa_rsync_changes_%s.csv" % datetime.now().strftime(
                 "%Y%m%d-%H%M%S"
```

Comment:
To improve readability
```suggestion
            outfilename = "ppa_rsync_changes_{time}.csv".format(
                time=datetime.now().strftime("%Y%m%d-%H%M%S")
```